### PR TITLE
Add a meaningful failure message when the CLI executable is not found.

### PR DESCRIPTION
**Pull request by gerrynelson63**
@dwstern or @tomstarr would you mind taking a look at this one as it impacts a number of tools.

For the tools the use the CLI the failure when it was not found in the location specified in application.properties was very cryptic. For those tools ,I added a function that checks that the CLI is in the specified location, and if it is not stops any further processing.  Thanks!

### CODE REVIEW

File: sharedfunctions.py [hunk 1]
```diff
@@ -881,3 +881,17 @@ def createdatefilter(days=0,datevar='creationTimeStamp',olderoryounger='older'):
...
+def getclicommand(execfile=""):
+
+    propertylist=getapplicationproperties()
+    clidir=propertylist["sascli.location"]
+    cliexe=propertylist["sascli.executable"]
+    clicommand=os.path.join(clidir,cliexe)
+
+    if not file_accessible(execfile,'r'):
+        print("ERROR: cannot find CLI at "+execfile+" check and update values in application.properties.")
+        clicommande=None
```
Comment (tomstarr) [inline sharedfunctions.py]: There is a typo in this variable's name.

File: createcaslibjson.py [hunk 2]
```diff
@@ -45,14 +45,6 @@
 import sys
 from sharedfunctions import callrestapi, getfolderid, file_accessible, printresult, getapplicationproperties
```
Comment (tomstarr) [inline createcaslibjson.py:46]: Should the getclicommand definition be included here?

File: getpath.py [hunk 3]
```diff
@@ -39,15 +39,6 @@
```
Comment (tomstarr) [inline getpath.py:39]: Should the getclicommand definition be included here?

File: getpathsplus.py [hunk 4]
```diff
@@ -57,14 +57,6 @@
 # get python version
```
Comment (tomstarr) [inline getpathsplus.py:57]: Should the getclicommand definition be included here?

File: testfolderaccess.py [hunk 5]
```diff
@@ -39,14 +39,6 @@
 # get python version
 version=int(str(sys.version_info[0]))
 
-# get cli location from properties
```
Comment (tomstarr) [inline testfolderaccess.py:42]: Should the getclicommand definition be being imported here?

Comment (tomstarr) [reply testfolderaccess.py:42]: And missing the lines that'd call it too?

```
# get cli location from properties, check that cli is there if not ERROR and stop
clicommand=getclicommand()
```

File: getpathsplus.py (hunk 4 above)
Comment (tomstarr) [reply getpathsplus.py:57]: And missing the lines that'd call it too?

```
# get cli location from properties, check that cli is there if not ERROR and stop
clicommand=getclicommand()
```

File: getpath.py (hunk 3 above)
Comment (tomstarr) [reply getpath.py:39]: And missing the lines that'd call it too?

```
# get cli location from properties, check that cli is there if not ERROR and stop
clicommand=getclicommand()
```

File: createcaslibjson.py (hunk 2 above)
Comment (tomstarr) [reply createcaslibjson.py:46]: And missing the lines that'd call it too?

```
# get cli location from properties, check that cli is there if not ERROR and stop
clicommand=getclicommand()
```

File: createbinarybackup.py [hunk 6]
```diff
@@ -34,14 +34,8 @@
 # get python version
```
Comment (tomstarr) [inline createbinarybackup.py:34]: Should the getclicommand definition be included here?

**Review by tomstarr — CHANGES_REQUESTED** [review]

File: sharedfunctions.py (hunk 1 above)
Comment (tomstarr) [reply sharedfunctions.py]: Resolving comment. I can see that has been resolved in a later commit.

**Review by tomstarr — APPROVED** [review]

File: requirements.txt [hunk 7]
```diff
@@ -1,14 +1,7 @@
...
-python-dateutil==2.8.2
-pytz==2022.1
-requests==2.31.0
-six==1.16.0
-swat==1.6.1
-urllib3==1.26.18
+configobj
+filedate
+numpy
+pandas
```
Comment (tomstarr) [inline requirements.txt:4]: Might be worth making a note about Pandas v2.1.2+ possibly not be fully supported by this project, until the way that it handles NaN values is fixed.
https://gitlab.sas.com/snztst/viyaauthmodel/-/issues/10
https://github.com/pandas-dev/pandas/issues/56379

**Comment by tomstarr** [comment]
Comments added throughout.
I've not tested the code from the branch, but the changes look good to go, once my comments have been review.

File: createcaslibjson.py (hunk 2 above)
Comment (gerrynelson63) [reply createcaslibjson.py:46]: That one doesn't actually use the CLI. I only added it to tools that actually make a CLI call

File: getpathsplus.py (hunk 4 above)
Comment (gerrynelson63) [reply getpathsplus.py:57]: only needed if we use the cli in the tool

File: requirements.txt (hunk 7 above)
Comment (gerrynelson63) [reply requirements.txt:4]: I will do that

**Pull request merged**